canonical.go

```diff
@@ -135,7 +131,10 @@ func (c *relaxedBodyCanonicalizer) Write(b []byte) (int, error) {
 		c.written = true
 	}
 
-	return c.w.Write(canonical)
+	if _, err := c.w.Write(canonical); err != nil {
+		return len(b), err
+	}
+	return len(b), nil
```

Comment:
You're right, `Write` should return the number of bytes written from `b`. However, you can simplify this to:

```go
_, err := c.w.Write(canonical)
return len(b), err
```

Reply:
yes, it's better and easier